clusterloader2/testing/load/config.yaml

```diff
@@ -15,28 +18,63 @@
 {{$bigRCsPerNamespace := DivideInt $bigRCs $namespaces}}
 {{$mediumRCsPerNamespace := DivideInt $mediumRCs $namespaces}}
 {{$smallRCsPerNamespace := DivideInt $smallRCs $namespaces}}
-{{$totalPodsRounded := AddInt (MultiplyInt $bigRCs $BIG_GROUP_SIZE) (MultiplyInt $mediumRCs $MEDIUM_GROUP_SIZE) (MultiplyInt $smallRCs $SMALL_GROUP_SIZE)}}
-# saturationQPS - qps for RC operations should be an equivalent to 10 [pods/s].
-{{$saturationQPS := DivideFloat (AddInt $bigRCs $mediumRCs $smallRCs) (DivideFloat $totalPodsRounded 10)}}
+{{$totalPods := AddInt (MultiplyInt $bigRCs $BIG_GROUP_SIZE) (MultiplyInt $mediumRCs $MEDIUM_GROUP_SIZE) (MultiplyInt $smallRCs $SMALL_GROUP_SIZE)}}
+# saturationTime - time for RC saturation should be an equivalent to 10 [pods/s] qps.
+{{$saturationTime := DivideFloat $totalPods $PODS_SATURATION_QPS}}
 #Test
 name: load
 automanagedNamespaces: {{$namespaces}}
 tuningSets:
-- name: AvgSaturationQPS
-  randomizedLoad:
-    averageQps: {{$saturationQPS}}
-- name: AvgScalingQPS
-  randomizedLoad:
+- name: SvcSaturationQPS
+  qpsLoad:
+    qps: {{$SVC_SATURATION_QPS}}
+- name: RandomizedSaturationTimeLimited
+  RandomizedTimeLimitedLoad:
+    timeLimit: {{MultiplyInt $saturationTime $SECOND_TO_NANO}}
+- name: RandomizedScalingTimeLimited
+  RandomizedTimeLimitedLoad:
     # The expected number of created/deleted pods is totalPods/4 when scaling,
     # as each RC changes its size from X to a uniform random value in [X/2, 3X/2].
-    # To match 10 [pods/s] requirement, we need to multiply saturationQPS by 4.
-    averageQps: {{MultiplyFloat $saturationQPS 4}}
+    # To match 10 [pods/s] requirement, we need to divide saturationTime by 4.
+    timeLimit: {{MultiplyInt (DivideFloat $saturationTime 4) $SECOND_TO_NANO}}
 steps:
 - measurements:
   - Identifier: APIResponsiveness
     Method: APIResponsiveness
     Params:
       action: reset
+# Create SVCs
+- phases:
+  - namespaceRange:
+      min: 1
+      max: {{$namespaces}}
+    replicasPerNamespace: {{$bigRCsPerNamespace}}
```

Comment:
We don't create service per RC currently:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scalability/load.go#L553

That said, apparently each pod is part of some service now.
So we should try:
- changing the current test to have service per RC
- if that works (and it should), then bundle service with RC creation together

Reply:
Services moved to the bundles.
Looks like fixing namespace problem (#256) and spreading creation randomly through the time, has fixed the performance issue.